tests/test_commander_core.py

```diff
@@ -270,16 +270,29 @@ def test_set_fixed_speed_error_commander_core(commander_core_device):
 
 def test_parse_channels_commander_core():
     """This test will go through and thoroughly test CommanderCore._parse_channels so we don't have to in other tests"""
-    tests = [
+    core = CommanderCore(MockCommanderCoreDevice(), 'Corsair Commander Core (experimental)', True)
+    corext = CommanderCore(MockCommanderCoreDevice(), 'Corsair Commander Core (experimental)', False)
+
+    core_tests = [
         ('pump', [0]), ('fans', [1, 2, 3, 4, 5, 6]),
         ('fan1', [1]), ('fan2', [2]), ('fan3', [3]), ('fan4', [4]), ('fan5', [5]), ('fan6', [6])
-             ]
+    ]
+
+    corext_tests = [
+        ('fans', [0, 1, 2, 3, 4, 5, 6]),
+        ('fan0', [0]), ('fan1', [1]), ('fan2', [2]), ('fan3', [3]), ('fan4', [4]), ('fan5', [5]), ('fan6', [6])
+    ]
 
-    for (val, answer) in tests:
-        assert list(CommanderCore._parse_channels(val)) == answer
+    # Run Core (non-XT) tests
+    for (val, answer) in core_tests:
+        assert list(core._parse_channels(val)) == answer
 
+    # Run Core XT tests
+    for (val, answer) in corext_tests:
+        assert list(corext._parse_channels(val)) == answer
```

Comment:
It would be better to split this into two tests, one for each variant.

This way, we get a better "at a glance" diagnostic if one of them ever fails, plus they are also better suited if, in the future, the implementations have to diverge more.

Reply:
Sure, makes sense. Updated.